hq/app/logic/InspectorResults.scala

```diff
@@ -45,26 +45,37 @@ object InspectorResults {
   }
 
   def parseDescribeAssessmentRunsResult(result: DescribeAssessmentRunsResult): List[InspectorAssessmentRun] = {
-    result.getAssessmentRuns.asScala.toList.map(parseAssessmentRun)
+    result.getAssessmentRuns.asScala.toList.flatMap(parseCompletedAssessmentRun)
   }
 
-  private[logic] def parseAssessmentRun(assessmentRun: AssessmentRun): InspectorAssessmentRun = {
-    InspectorAssessmentRun(
-      arn = assessmentRun.getArn,
-      name = assessmentRun.getName,
-      appId = InspectorResults.appId(assessmentRun.getName),
-      assessmentTemplateArn = assessmentRun.getAssessmentTemplateArn,
-      state = assessmentRun.getState,
-      durationInSeconds = assessmentRun.getDurationInSeconds,
-      rulesPackageArns = assessmentRun.getRulesPackageArns.asScala.toList,
-      userAttributesForFindings = assessmentRun.getUserAttributesForFindings.asScala.toList.map(attr => (attr.getKey, attr.getValue)),
-      createdAt = new DateTime(assessmentRun.getCreatedAt),
-      startedAt = new DateTime(assessmentRun.getStartedAt),
-      completedAt = new DateTime(assessmentRun.getCompletedAt),
-      stateChangedAt = new DateTime(assessmentRun.getStateChangedAt),
-      dataCollected = assessmentRun.getDataCollected,
-      findingCounts = assessmentRun.getFindingCounts.asScala.toMap.mapValues(_.toInt)
-    )
+  /**
+    * Parses a *completed* assessment, if it matches the format used by our automatic inspection service.
+    */
+  private[logic] def parseCompletedAssessmentRun(assessmentRun: AssessmentRun): Option[InspectorAssessmentRun] = {
+    if (assessmentRun.getState != "COMPLETED" || assessmentRun.getDataCollected != true) {
```

Comment:
if (!= || !=) A else B

Personally I'd rewrite to the opposite clause if (== && ==) B else A

Reply:
Great shout, thanks